posts/blundell/You_have_permission_to.jade

```diff
@@ -0,0 +1,204 @@
+---
+title:      'You have permission to crash'
+categories: [build]
+heroimage:  'crash-hero.png'
+excerpt:    'With targetting Marshmallow, 6.0, API 23, the latest version of Android, strange things can happen if you dont get your permission flow correct.' 
+---
+
+extends ../../views/layouts/_single
+
+block content
+  :marked
+    ### You have permission to crash
+
+    With targetting Marshmallow, 6.0, API 23, the latest version of Android, strange things can happen if you don't get your permission flow correct. The code seems legit and yet it crashes every time. You have all the correct permissions in your AndroidManifest, you call the correct api to get your data. It runs perfectly well on your other emulator (_pre-marshmallow without realising_). Let's debug together.
+
+    Permissions are split into two categories, _dangerous_ and _normal_ the rest of this article is implicitly talking about _dangerous_ permissions.
+    You can read more about the [difference here.](http://developer.android.com/guide/topics/security/permissions.html#normal-dangerous) 
+
+    > "Once you eliminate the impossible, whatever remains, no matter how improbable, must be the truth."
+    >                                                           - Arthur Conan Doyle
+
+    As a good developer you've rechecked the code thrice, you've rebooted the emulator, hell you even closed Android Studio and re-opened it, _did you try Eclipse ... I hope you're not that desperate_! What else has changed... of course you're running against a new version of Android. 
```

Comment:
`What else has changed... of course you're running against a new version of Android...`

->

`What else has changed..? Of course! You're running against a new version of Android...`
`What else has changed..? Of course, you're running against a new version of Android!`
`What else has changed..? Of course, you're running against a new version of Android..!`